packages/dbml-connector/src/connectors/mssqlConnector.ts

```diff
@@ -49,19 +49,51 @@ const getValidatedClient = async (connection: string): Promise<sql.ConnectionPoo
 
 const convertQueryBoolean = (val: string | null): boolean => val === 'YES';
 
-const getFieldType = (data_type: string, default_type: DefaultType, character_maximum_length: number, numeric_precision: number, numeric_scale: number): string => {
+const getFieldType = (data_type: string, character_maximum_length: number, numeric_precision: number, numeric_scale: number): string => {
   if (MSSQL_DATE_TYPES.includes(data_type)) {
     return data_type;
   }
+
+  // timestamp is a synonym of rowversion
+  // microsoft recommends that we should not use timestamp, so we will convert it back to rowversion
+  // https://learn.microsoft.com/en-us/sql/t-sql/data-types/rowversion-transact-sql?view=sql-server-ver15
+  if (data_type === 'timestamp') {
+    return 'rowversion';
+  }
+
+  // process numeric -based type
   if (data_type === 'bit') {
     return data_type;
   }
-  if (numeric_precision && numeric_scale && default_type === 'number') {
-    return `${data_type}(${numeric_precision},${numeric_scale})`;
+  // if precision != 0 => numeric-based column
+  if (numeric_precision) {
+    return numeric_scale
+      ? `${data_type}(${numeric_precision},${numeric_scale})`
+      : `${data_type}(${numeric_precision})`;
   }
-  if (character_maximum_length && character_maximum_length > 0 && default_type === 'string') {
-    return `${data_type}(${character_maximum_length})`;
+
+  // process string-based type
+  // ntext, text & image
+  if (['ntext', 'text', 'image'].includes(data_type)) {
+    return data_type;
   }
+
+  // Column data type is varchar(max), nvarchar(max), varbinary(max), or xml
+  if (character_maximum_length < 0) {
+    return data_type === 'xml' ? data_type : `${data_type}(MAX)`;
+  }
+
+  // character_maximum_length is the lenght in bytes
+  // nchar and nvarchar store Unicode characters, each character needs 2 bytes
+  // so we have to divide it by 2 to get the correct maximum lenght in character.
+  if (character_maximum_length > 0) {
+    const maximum_length_in_character = (data_type === 'nchar' || data_type === 'nvarchar')
+      ? character_maximum_length / 2
+      : character_maximum_length;
+
+    return `${data_type}(${maximum_length_in_character})`;
+  }
```

Comment:
Looks good, please also add the url to the mssql documentation related to this re-calculation em.

Reply:
Added a link in https://github.com/holistics/dbml/pull/645/commits/74b7e23ae137b61b30ed2ffdb2bbb5f9c727ad14